packages/components/src/Offer/Form/Checkout.tsx

```diff
@@ -151,6 +149,21 @@ const OfferFormCheckout: FC<Props> = ({
                 placeholder={t('offer.form.checkout.quantity.placeholder')}
                 {...register('quantity', {
                   required: t('offer.form.checkout.validation.required'),
+                  validate: (value) => {
+                    if (
+                      parseInt(value, 10) >= 0 &&
+                      parseInt(value, 10) <=
+                        parseInt(offer.availableQuantity, 10)
+                    ) {
+                      if (!/^\d+$/.test(value)) {
+                        return t('offer.form.checkout.validation.integer')
+                      }
+                    } else {
+                      return t('offer.form.checkout.validation.in-range', {
+                        available: parseInt(offer.availableQuantity, 10),
+                      })
+                    }
+                  },
```

Comment:
Would be a lot better to do the conditions to check for wrong value rather than checking for good value and have to cascade them together:
```suggestion
                  validate: (value) => {
                    if (!/^\d+$/.test(value)) {
                      return t('offer.form.checkout.validation.integer')
                    }
                    if (
                      parseInt(value, 10) < 0 ||
                      parseInt(value, 10) > parseInt(offer.availableQuantity, 10)
                    ) {
                      return t('offer.form.checkout.validation.in-range', {
                        available: parseInt(offer.availableQuantity, 10),
                      })
                    }
                  },
```